packages/build-tools/src/steps/functions/repack.ts

```diff
@@ -0,0 +1,187 @@
+import fs from 'node:fs';
+import os from 'node:os';
+import path from 'node:path';
+
+import { type Job, Platform } from '@expo/eas-build-job';
+import { type bunyan } from '@expo/logger';
+import {
+  BuildFunction,
+  BuildStepInput,
+  BuildStepInputValueTypeName,
+  BuildStepOutput,
+  spawnAsync,
+} from '@expo/steps';
+import {
+  repackAppAndroidAsync,
+  repackAppIosAsync,
+  type Logger,
+  type SpawnProcessAsync,
+  type SpawnProcessOptions,
+  type SpawnProcessPromise,
+  type SpawnProcessResult,
+} from '@expo/repack-app';
+import { v4 as uuidv4 } from 'uuid';
+
+import { pathExistsAsync } from '../../utils/files';
+
+export function createRepackBuildFunction(): BuildFunction {
+  return new BuildFunction({
+    namespace: 'eas',
+    id: 'repack',
+    name: 'Repack app',
+    inputProviders: [
+      BuildStepInput.createProvider({
+        id: 'source_app',
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+        required: true,
+      }),
+      BuildStepInput.createProvider({
+        id: 'output_path',
+        allowedValueTypeName: BuildStepInputValueTypeName.STRING,
+        required: false,
+      }),
+      BuildStepInput.createProvider({
```

Comment:
I think there's a `working_directory` param on steps themselves? https://github.com/expo/eas-build/blob/9a804e6f60e484fee8645bcb640ddd8756f7a0fd/packages/steps/src/BuildConfig.ts#L187 (As strange as it feels to have a working directory for a function)